lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -955,6 +1033,29 @@ def _upgrade_opensearch(self, event: _UpgradeOpenSearch) -> None:
                 return
         logger.debug("Acquired lock for upgrade")
 
+        # TODO upgrade: catch http errors
+
+        # TODO upgrade: does this affect user set values?
+        # TODO look at `preserve_existing`
+        # https://www.elastic.co/guide/en/elasticsearch/reference/8.13/indices-update-settings.html
+
+        # Increase timeout before shard allocation replicates shards on offline (e.g. upgrading)
+        # node to other nodes
+        # Used instead of disabling shard allocation entirely
+        # (Less dangerous if this unit fails to upgrade & reset the timeout—it's safer to have
+        # increased timeout than shard allocation disabled entirely, especially for long-running
+        # upgrades)
+        # Replaces "Disable shard allocation" step from
+        # https://www.elastic.co/guide/en/elastic-stack/8.13/upgrading-elasticsearch.html#rolling-upgrades
+        self.opensearch.request(
+            "PUT",
+            "/_all/_settings",
+            payload={"settings": {"index.unassigned.node_left.delayed_timeout": "5m"}},
+        )
```

Comment:
This will fail, because of the inclusion of the .opendistro_security index in `_all` - which cannot be altered... 
We should be more cautious here, and change this to the following:
```suggestion
        indices = self.opensearch.request("GET", "/_settings?flat_settings=true")
        indices_to_update = ",".join([
            index
            for index, settings in indices.items()
            if not index.startswith(".") and "index.unassigned.node_left.delayed_timeout" not in settings
        ])
        if indices_to_update:
            self.opensearch.request(
                "PUT",
                f"/{indices_to_update}/_settings",
                payload={"settings": {"index.unassigned.node_left.delayed_timeout": 5m}},
            )
            
            # TODO: either store `indices_to_update` this in relation data 
            # OR: store them in an internal index.
```